chrysom/listenerClient.go

```diff
@@ -38,88 +39,108 @@ const (
 )
 
 // ListenerConfig contains config data for polling the Argus client.
-type ListenerClientConfig struct {
-	// Listener provides a mechanism to fetch a copy of all items within a bucket on
-	// an interval.
+type ListenerClientIn struct {
+	fx.In
+
+	// Listener fetches a copy of all items within a bucket on
+	// an interval based on `BasicClientConfig.PullInterval`.
 	// (Optional). If not provided, listening won't be enabled for this client.
 	Listener Listener
-
-	// PullInterval is how often listeners should get updates.
-	// (Optional). Defaults to 5 seconds.
-	PullInterval time.Duration
```

Comment:
move `PullInterval` to BasicClientConfig in order to simplify setup and fx dependencies